openedx/core/djangoapps/team_api/views.py

```diff
@@ -0,0 +1,407 @@
+            try:
+                course_key = CourseKey.from_string(request.QUERY_PARAMS['course_id'])
+                result_filter.update({'course_id': course_key})
+            except InvalidKeyError:
+                return Response({'detail': "course_id is not valid"}, status=status.HTTP_400_BAD_REQUEST)
+        if 'topic_id' in request.QUERY_PARAMS:
+            result_filter.update({'topic_id': request.QUERY_PARAMS['topic_id']})
+        if 'include_inactive' in request.QUERY_PARAMS and request.QUERY_PARAMS['include_inactive'].lower() == 'true':
+            del result_filter['is_active']
+        if 'text_search' in request.QUERY_PARAMS:
+            return Response({'detail': "text_search is not yet supported"}, status=status.HTTP_400_BAD_REQUEST)
+
+        queryset = CourseTeam.objects.filter(**result_filter)
+        queryset = queryset.extra(select={'lower_name': "lower(name)"})
+
+        order_by_field = 'lower_name'
+        if 'order_by' in request.QUERY_PARAMS:
+            order_by_input = request.QUERY_PARAMS['order_by']
+            if order_by_input == 'name':
+                order_by_field = 'lower_name'
+            elif order_by_input == 'open_slots':
+                queryset = queryset.annotate(team_size=Count('users'))
+                order_by_field = 'team_size'
+            elif order_by_input == 'last_activity':
+                return Response({'detail': "last_activity is not yet supported"}, status=status.HTTP_400_BAD_REQUEST)
+
+        queryset = queryset.order_by(order_by_field)
+
+        if not queryset:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        page = self.paginate_queryset(queryset)
+        serializer = self.get_pagination_serializer(page)
+        return Response(serializer.data)
+
+    def post(self, request):
+        """
+        POST /api/team/v0/teams/
+        """
+        field_errors = {}
+        course_key = None
+
+        try:
+            course_key = CourseKey.from_string(request.DATA.get('course_id'))
+            get_course(course_key)
+        except InvalidKeyError:
+            field_errors['course_id'] = {
+                'developer_message': "course_id is not valid.",
+            }
+        except ValueError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        for key in request.DATA:
+            if key not in ['name', 'course_id', 'description', 'topic_id', 'country', 'language']:
+                field_errors[key] = {
+                    'developer_message': "This field is not present on this resource",
+                }
+
+        team = CourseTeam.create(
+            name=request.DATA.get('name', ""),
+            course_id=course_key,
+            description=request.DATA.get('description', ""),
+            topic_id=request.DATA.get('topic_id'),
+            country=request.DATA.get('country'),
+            language=request.DATA.get('language'),
+        )
+
+        try:
+            team.full_clean()
```

Comment:
Instead of doing full_clean, I think you should be calling is_valid. Take a look at _add_serializer_errors in /edx-platform/openedx/core/djangoapps/user_api/accounts/api.py. It might be good to generalize this method as a helper method.

Related to this, I would include user_messages too.